### PR TITLE
round balance and preview

### PR DESCRIPTION
**Pull request by vanillaHill**
Closes https://github.com/FortressFinance/issues/issues/122
closes https://github.com/FortressFinance/issues/issues/123

### CODE REVIEW

File: src/components/TokenForm/TokenForm.tsx [hunk 1]
```diff
@@ -143,7 +144,7 @@ const TokenForm: FC<TokenFormProps> = ({
           <span>
             {!isWithdraw ? "Balance: " : "Share: "}
             <Skeleton isLoading={isLoadingInputTokenBalanceOrShare}>
-              {inputTokenBalanceOrShare?.formatted ?? "0.0"}
+              {toFixed(inputTokenBalanceOrShare?.formatted ?? "0.0", 2)}
```
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/components/TokenForm/TokenForm.tsx:147]: It's just an arbitrary value, I think 2 decimals is enough for the balance/share. Will change it to 6 to normalize everything.

File: src/lib/api/util/format.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  rounding: Rounding = Rounding.ROUND_HALF_UP
+): string => {
+  invariant(!isNaN(Number(value)), `${value} is not an number.`)
+  invariant(
+    Number.isInteger(decimalPlaces),
+    `${decimalPlaces} is not an integer.`
+  )
+  invariant(decimalPlaces >= 0, `${decimalPlaces} is negative.`)
+
+  Big.DP = decimalPlaces
```
Comment (aquaticone) [inline src/lib/api/util/format.ts:61]: I think you can just do something simple like this to round to 4 decimal places:

```ts
// value = 824213281784279560
const formattedValue = ethers.utils.formatEther(value);
// formattedValue = 0.82421328178427956
return Math.round(res * 1e4) / 1e4;
// return 0.8242
```

**Review by aquaticone — CHANGES_REQUESTED** [review]
We should avoid adding multiple libraries for working with big numbers.

File: src/lib/api/util/format.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import _Big from "big.js"
+import _Decimal from "decimal.js-light"
```
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/lib/api/util/format.ts:2]: it's a light version of the decimal

File: src/lib/api/util/format.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  [Rounding.ROUND_UP]: Big.roundUp,
+}
+
+export const toSignificant = (
+  value: string,
+  significantDigits: number,
+  format: object = { groupSeparator: "" },
+  rounding: Rounding = Rounding.ROUND_HALF_UP
+): string => {
+  invariant(!isNaN(Number(value)), `${value} is not an number.`)
```
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/lib/api/util/format.ts:33]: Interesting

File: src/lib/api/util/format.ts (hunk 2 above)
Comment (vanillaHill) [reply src/lib/api/util/format.ts:61]: We can certainly do it Math.round for this particular case but the method I'm trying to add here is something generic that we can use for any number format we could meet inside the project. Bring more flexibility, specify the group or the decimal separators, change the decimal number etc in that method.

File: src/lib/api/util/format.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (aquaticone) [reply src/lib/api/util/format.ts:33]: You don't need to switch now, its OK. We can always update in the future if we end up needing more validation.

File: src/lib/api/util/format.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import _Big from "big.js"
```
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/lib/api/util/format.ts]: Removed

File: src/lib/api/util/format.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/lib/api/util/format.ts]: Done

File: src/lib/api/util/format.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/lib/api/util/format.ts]: Removed

**Review by aquaticone — APPROVED** [review]
LGTM

**Pull request merged**